compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/lower_to_aie.mlir

```diff
@@ -300,26 +300,37 @@ module {
 
 // -----
 
+// Test to show mix of implicit/explicit source/target addressing in amdaie.npu.dma_cpy_nd.
+
 // CHECK:       aie.device
 // CHECK-DAG:   %[[TILE_0_2:.+]] = aie.tile(0, 2)
 // CHECK-DAG:   %[[TILE_0_1:.+]] = aie.tile(0, 1)
 // CHECK-DAG:   %[[TILE_0_0:.+]] = aie.tile(0, 0)
 // CHECK:       aie.objectfifo @[[OBJ0:.+]](%[[TILE_0_2]], {%[[TILE_0_1]]}
 // CHECK-NEXT:  aie.objectfifo @[[OBJ1:.+]](%[[TILE_0_1]], {%[[TILE_0_0]]}
-// CHECK-NEXT:  aie.objectfifo.link
-// CHECK-SAME:  @[[OBJ0]]
-// CHECK-SAME:  @[[OBJ1]]
-// CHECK:       func.func @controlcode
-// CHECK-SAME:  %[[ARG0:.+]]: memref<32x64xi32>
+// CHECK-NEXT:  aie.objectfifo @[[OBJ2:.+]](%[[TILE_0_0]], {%[[TILE_0_1]]}
+// CHECK:       aie.objectfifo.link [@[[OBJ0]]] -> [@[[OBJ1]]]
+// CHECK:       func.func @controlcode(%[[ARG0:.+]]: memref<32x64xi32>
 // CHECK:         aiex.npu.dma_memcpy_nd
-// CHECK-SAME:    %[[ARG0]]
-// CHECK-SAME:    [1, 1, 0, 32]
-// CHECK-SAME:    [1, 1, 32, 32]
-// CHECK-SAME:    [1, 1, 64]
-// CHECK-SAME:    issue_token = true
-// CHECK-SAME:    @[[OBJ1]]
-// CHECK-NEXT:    aiex.npu.dma_wait
-// CHECK-SAME:    @[[OBJ1]]
+// CHECK-SAME:            %[[ARG0]][1, 1, 0, 32][1, 1, 32, 32][1, 1, 64]
+// CHECK-SAME:            issue_token = true
+// CHECK-SAME:            metadata = @[[OBJ1]]
+// CHECK-NEXT:    aiex.npu.dma_wait {symbol = @[[OBJ1]]}
+// CHECK:         aiex.npu.dma_memcpy_nd
+// CHECK-SAME:            %[[ARG0]][0, 0, 0, 0][0, 0, 32, 64][2048, 2048, 64]
```

Comment:
Hi @jtuyls - could you check if it looks okay to you?

The former one is for the explicit target addressing whereas the latter for implicit.

Similarly for the two cases below for source addressing.

Reply:
Yes, this output looks good